.github/ISSUE_TEMPLATE/upstream-sync.md

```diff
@@ -0,0 +1,123 @@
+---
+name: Upstream sync
+about: Outlines the process to synchonize ODH fork with upstream repositories
+title: "[MM] Synchronization with upstream repositories"
+labels: ''
+assignees: ''
+
+---
+
+A ModelMesh synchronization involves four repositories: _modelmesh_ , _modelmesh-runtime-adapter_, _rest-proxy_ and _modelmesh-serving_.
+
+ModelMesh depends on some KServe resources. You may want to synchronize [KServe fork](https://github.com/opendatahub-io/kserve) first and, secondly, ModelMesh.
+
+## Code synchronization
+
+The steps for doing code synchronization are the same for all repositories. You can use the following script on an empty folder as a helper (it assumes you have your own forks using same names as in ODH):
+
+```sh
+export GH_USER_NAME=$USER       # Replace with your GitHub handle
+export SYNC_HOME=$(mktemp -d)   # Either keep as is, or set to an empty directory of your liking
+export REPOS_TO_SYNC=(modelmesh modelmesh-runtime-adapter rest-proxy modelmesh-serving)
+export RELEASE_BRANCH_NAME=release-0.11
+export SYNC_BRANCH_NAME=${RELEASE_BRANCH_NAME}_$(date '+%Y%m%d')_sync
+
+cd $SYNC_HOME
+
+for TARGET_REPO_NAME in ${REPOS_TO_SYNC[@]}
+do
+  git clone git@github.com:${GH_USER_NAME}/${TARGET_REPO_NAME}.git
+  pushd ${TARGET_REPO_NAME}/
+
+  git remote add odh git@github.com:opendatahub-io/${TARGET_REPO_NAME}.git
+  git remote add kserve https://github.com/kserve/${TARGET_REPO_NAME}
+
+  git fetch odh
+  git fetch kserve
+
+  git fetch odh ${RELEASE_BRANCH_NAME}
+  git fetch kserve ${RELEASE_BRANCH_NAME}
+
+  # Sync release branch
+  git branch ${SYNC_BRANCH_NAME} odh/${RELEASE_BRANCH_NAME}
+  git checkout ${SYNC_BRANCH_NAME}
+  git merge kserve/${RELEASE_BRANCH_NAME}
+
+  popd
+done
+
+echo ""
+echo "Sync directory: $SYNC_HOME"
+
+for TARGET_REPO_NAME in ${REPOS_TO_SYNC[@]}
+do
+  is_clean=$(cd $TARGET_REPO_NAME; git status --porcelain)
+  if [ -z "$is_clean" ]; then
+    echo "$TARGET_REPO_NAME: Clean"
+  else
+    echo "$TARGET_REPO_NAME: Conflict"
+  fi
+done
+```
+
+When the script finishes, you will see a sumary about merge status. You will need to manually fix any conflicts and finish the merge commiting the changes `git commit -s -m "Sync upstream ${RELEASE_BRANCH_NAME}"`
+
+Once all repositories are synced, you can push changes to your own fork and create PRs for all repositories, except `modelmesh-serving`.
```

Comment:
As we discussed it earlier, can you please rephrase this part?

Reply:
@spolti I just updated the PR, FWIW.
Not sure if we still want to merge this.

Reply:
Let's move to another place, it is indeed good content to serve as a doc.

Reply:
I just moved to `opendatahub/docs` hoping it is better.